catalogbuilder/intakebuilder/gfdlcrawler.py

```diff
@@ -114,12 +117,26 @@ def crawlLocal(projectdir, dictFilter,dictFilterIgnore,logger,configyaml,slow):
                # todo do the reverse if slow is on. Open file no matter what and populate dictionary values and if there is something missed out
                # we can scan filenames or config etc 
                #here, we will see if there are missing header values and compare with file attributes if slow option is turned on
-               if (slow == True) & (bool(dictInfo) == True) :
-                    print("Slow option turned on.. lets open some files using xarray and lookup atts",filename)
-                    #todo we could look at var attributes, but right now we stick to those that are necessary. scope to extend this easily to missngcols or if header info is not in config yaml 
-                    if "standard_name" in missingcols: 
+               if (slow == True) & (bool(dictInfo) == True):
+                    #print("Slow option turned on.. lets open some files using xarray and lookup atts")
+                    
+                    #todo we could look at var attributes, but right now we stick to those that are necessary. scope to extend this easily to missngcols or if header info is not in config yaml
+                    if "standard_name" in missingcols:
+
+                        # Set standard_name as na to avoid error from getInfoFromVarAtts
                         dictInfo["standard_name"] = "na"
-                        getinfo.getInfoFromVarAtts(dictInfo["path"],dictInfo["variable_id"],dictInfo)
+
+                        # qualities define the uniqueness and help us determine when to open files. here, we define uniqueness by realm and var_id combinations. we store the realm/var_id pairs + their standard_names in unique_datasets{} and the current pair being checked as a tuple list called 'qualities'. if a pair stored in unique_datasets aligns with the current pair being checked, we won't open the file and will instead use the standard_name already found
+                        qualities=(dictInfo["variable_id"],dictInfo["realm"])
```

Comment:
Add a TODO to remind us to extend this to append other qualities. No code changes needed now. This looks good!

I also wonder for a future issue, if we can only open the files if the fast option look up table misses the standard name. This would use slow and fast! Again a TODO. If you think this is reasonable, open an issue to just table this for a future date.

Reply:
```suggestion
                        #TODO extend this to append other qualities later
                        qualities=(dictInfo["variable_id"],dictInfo["realm"])
```